apps/foundation-demo/src/app/modals/cart-modal/components/cart-modal/cart-modal.component.html

```diff
@@ -0,0 +1,15 @@
+<daff-modal [show]="show$ | async" (hide)="onHide()" [verticalPosition]="_verticalPosition" [horizontalPosition]="_horizontalPosition">
+  <div class="cart-modal" cart-container #CartContainer="CartContainer">
+    <div class="cart-modal__top-bar">
+      <span class="cart-modal__check-icon"><i class="fa fa-check" aria-hidden="true"></i></span>
+      <span>added to cart</span>
+      <span class="cart-modal__times-icon"><i (click)="onHide()" class="fa fa-times" aria-hidden="true"></i></span>
+    </div>
+    <checkout-cart class="cart-modal__checkout-cart" [cart]="CartContainer.cart$ | async" *ngIf="!(CartContainer.loading$ | async)"></checkout-cart>
+    <loading-icon *ngIf="CartContainer.loading$ | async"></loading-icon>
+    <div class="cart-modal__buttons">
```

Comment:
`daff-button-set` only works if the children are buttons, right? I think `daff-button-set` needs to be changed a bit so the children can be any element, because most of the buttons will probably be wrapped in a component

Reply:
I'll make updates to the `daff-button-set` in a new PR and update this file then.

Reply:
Addressed in #100

Reply:
So it still doesn't work as expected, because `button`s act like `span`s and line up horizontally. But most html elements just stack on top of each other. So
```
<div daff-button-set>
  <some-button-component></some-button-component>
  <some-button-component></some-button-component>
</div>
```
doesn't work

Reply:
Because of Separation of Concerns. Here's a good explanation: https://softwareengineering.stackexchange.com/questions/32581/how-do-you-explain-separation-of-concerns-to-others

Reply:
Okay, so I had to add some display properties to the `daff-button-set` component and fixed the flex issue. I think separation of concerns is a valid point, but also not sure if it's really needed for the buttons. @damienwebdev thoughts?

Reply:
If `proceed-to-checkout` wasn't its own component, we'd already have duplicate code in both the `cart-modal` and the `cart`. The html and this function would be duplicated: 
```
navigateToCheckout() {
  this.router.navigateByUrl('/checkout');
}
```